assets/js/base/components/product-list/index.js

```diff
@@ -10,20 +10,58 @@ import classnames from 'classnames';
 import Pagination from '@woocommerce/base-components/pagination';
 import ProductSortSelect from '@woocommerce/base-components/product-sort-select';
 import ProductListItem from '@woocommerce/base-components/product-list-item';
-import withProducts from '@woocommerce/base-hocs/with-products';
+import {
+	useStoreProducts,
+	useSynchronizedQueryState,
+} from '@woocommerce/base-hooks';
 import withScrollToTop from '@woocommerce/base-hocs/with-scroll-to-top';
 import './style.scss';
 
+const generateQuery = ( { sortValue, currentPage, attributes } ) => {
+	const { columns, rows } = attributes;
+	const getSortArgs = ( orderName ) => {
+		switch ( orderName ) {
+			case 'menu_order':
+			case 'popularity':
+			case 'rating':
+			case 'date':
+			case 'price':
+				return {
+					orderby: orderName,
+					order: 'asc',
+				};
+			case 'price-desc':
+				return {
+					orderby: 'price',
+					order: 'desc',
+				};
+		}
+	};
+	return {
+		...getSortArgs( sortValue ),
+		per_page: columns * rows,
+		page: currentPage,
+	};
+};
+
 const ProductList = ( {
 	attributes,
 	currentPage,
 	onPageChange,
 	onSortChange,
 	sortValue,
-	products,
 	scrollToTop,
-	totalProducts,
 } ) => {
+	// @todo query-state context (which is hardcoded 'product-grid' here) should
+	// be provided via a provider. This allows us to control query state context
+	// at the app level for anything nested within the provider.
+	const [ queryState ] = useSynchronizedQueryState(
+		'product-grid',
```

Comment:
Perhaps we should default to a state for the entire page (`global-product-query`) and then allow groups of blocks (say a grid and a price filter) within a component to be passed a custom context so they work independently.

That would allow groups of blocks to work independently if needed, and allow any blocks not in a group to share query states without affecting the groups.

Does that even make sense :D

Reply:
Kind of, but if I'm understanding it correctly, that's further support for having Context for query state.